### PR TITLE
test/ffmpeg: add vpp/csc r2r test

### PR DESCRIPTION
**Pull request by Hangjie22Coder**

### CODE REVIEW

File: test/ffmpeg-qsv/vpp/csc.py [hunk 1]
```diff
@@ -24,3 +25,10 @@ def test(self, case, csc):
     vars(self).update(spec[case].copy())
     vars(self).update(case = case, csc = csc)
     self.vpp()
+
+  @slash.parametrize(*gen_vpp_csc_parameters(spec_r2r))
+  def test_r2r(self, case, csc):
+    vars(self).update(spec[case].copy())
+    vars(self).update(case = case, csc = csc)
+    vars(self).setdefault("r2r", 5)
+    self.vpp()
```
Comment (uartie) [inline test/ffmpeg-qsv/vpp/csc.py:34]: Please add empty new line at end of file.

File: test/ffmpeg-vaapi/vpp/csc.py [hunk 2]
```diff
@@ -25,3 +26,10 @@ def test(self, case, csc):
     vars(self).update(spec[case].copy())
     vars(self).update(case = case, csc = csc)
     self.vpp()
+
+  @slash.parametrize(*gen_vpp_csc_parameters(spec_r2r))
+  def test_r2r(self, case, csc):
+    vars(self).update(spec[case].copy())
+    vars(self).update(case = case, csc = csc)
+    vars(self).setdefault("r2r", 5)
+    self.vpp()
```
Comment (uartie) [inline test/ffmpeg-vaapi/vpp/csc.py:35]: Please add empty new line at end of file.

File: test/gst-msdk/vpp/csc.py [hunk 3]
```diff
@@ -24,3 +25,10 @@ def test(self, case, csc):
     vars(self).update(spec[case].copy())
     vars(self).update(case = case, csc = csc)
     self.vpp()
+
+  @slash.parametrize(*gen_vpp_csc_parameters(spec_r2r))
+  def test_r2r(self, case, csc):
+    vars(self).update(spec[case].copy())
+    vars(self).update(case = case, csc = csc)
+    vars(self).setdefault("r2r", 5)
+    self.vpp()
```
Comment (uartie) [inline test/gst-msdk/vpp/csc.py:34]: Please add empty new line at end of file.

File: test/gst-va/vpp/csc.py [hunk 4]
```diff
@@ -24,3 +25,10 @@ def test(self, case, csc):
     vars(self).update(spec[case].copy())
     vars(self).update(case = case, csc = csc)
     self.vpp()
+
+  @slash.parametrize(*gen_vpp_csc_parameters(spec_r2r))
+  def test_r2r(self, case, csc):
+    vars(self).update(spec[case].copy())
+    vars(self).update(case = case, csc = csc)
+    vars(self).setdefault("r2r", 5)
+    self.vpp()
```
Comment (uartie) [inline test/gst-va/vpp/csc.py:34]: Please add empty new line at end of file.

File: test/gst-vaapi/vpp/csc.py [hunk 5]
```diff
@@ -24,3 +25,10 @@ def test(self, case, csc):
     vars(self).update(spec[case].copy())
     vars(self).update(case = case, csc = csc)
     self.vpp()
+
+  @slash.parametrize(*gen_vpp_csc_parameters(spec_r2r))
+  def test_r2r(self, case, csc):
+    vars(self).update(spec[case].copy())
+    vars(self).update(case = case, csc = csc)
+    vars(self).setdefault("r2r", 5)
+    self.vpp()
```
Comment (uartie) [inline test/gst-vaapi/vpp/csc.py:34]: Please add empty new line at end of file.

**Review by uartie — CHANGES_REQUESTED** [review]
Please split your changes into multiple commits... one for each middleware.

Additionally, commit messages should always have an empty line between the title and the message body.  Otherwise, mulitple lines in title will be concat as one line in git logs.... and that makes them hard to read.

**Comment by Hangjie22Coder** [comment]
> Please split your changes into multiple commits... one for each middleware.
> 
> Additionally, commit messages should always have an empty line between the title and the message body. Otherwise, mulitple lines in title will be concat as one line in git logs.... and that makes them hard to read.

Thanks for the review, updated.

**Review by uartie — APPROVED** [review]

**Pull request merged**